### PR TITLE
Changes to add all transformations to tracker

### PR DESCRIPTION
**Pull request by BrandonNowlin**
NuRelic Changes: 
Big Picture:
    Added all transformations to the tracker and gave them variables that can be referenced for the description on the user-side
        Added "spun" and "leviathan" to the default status message
    Added Transformations:
        Bob
        Conjoined
        Fun Guy
        Leviathan
        Oh Crap
        Seraphim
        Spun
        Yes, Mother
        Super Bum
        Beelzebub

Specifics:
    options.json:
        Added "/ Leviathan: {leviathan} / Spun: {spun}" 
    options_default.json:
        Added "/ Leviathan: {leviathan} / Spun: {spun}" 
    overlay text reference (Folder):
        Added transformation files, defaulted them to 0
    src > game_objects > item.py:
        Added transformations to comment pertaining to properties accessible
    src > game_objects > state.py:
        Added: 
            One set for each transformation
            Check each item for a transformation, if it is one of them, add that item to the set
    src > view_controls > overlay.py
        Added:
            Check on update_stats to check for each transformation
            Changed format_guppy to format_transformation and changed calls to function in both overlay.py and view.py
    src > view_controls > view.py
        Added (to keep comments accurate): 
            # Draw seed/transformation text:
            Fill the dictionary with each transformation 
            Add check in "shown" for each transformation
    items.json:
        Beelzebub Added with "beelzebub": true to the following:
            ???'s Only Friend (320)
            BBF (272)
            Best Bud (274)
            Big Fan (279)
            Distant Admiration (57)
            Forever Alone (128)
            Halo Of Flies (10)
            Hive Mind (248)
            Skatole (9)
            Smart Fly (264)
            The Mulligan (151)
            Infestation (148)
            Friend Zone (364)
            Lost Fly (365)
            Papa Fly (430)
            Obsessed Fan(426)
        Bob Added with "bob": true to the following:
            Bob's Rotten Head (42)
            Bob's Brain (273)
            Bob's Curse (140)
            Ipecac (149)
        Conjoined Added with "conjoined": true to the following:
            Brother Bobby (8)
            Harlequin Baby (167)
            Headless Baby (269)
            Little Steven (100)
            Mongo Baby (322)
            Rotten Baby (268)
            Sister Maggy (67)
        Fun Guy Added with "funguy": true to the following:
            God's Flesh (398)
            Mini Mush (71)
            Magic Mushroom (12)
            Odd Mushroom (Thin) (120)
            Odd Mushroom (Large) (121)
            1up! (11)
            Blue Cap (342)
        Leviathan Added with "leviathan": true to the following:
            The Nail (83)
            The Mark (79)
            The Pact (80)
            Pentagram (51)
            Spirit Of The Night (159)
            Maw Of The Void (399)
            Abaddon (230)
            Brimstone (118)
        Oh Crap Added with "ohcrap": true to the following:
            Flush! (291)
            The Poop (36)
            E. Coli (236)
        Seraphim Added with "seraphim": true to the following:
            The Bible (33)
            Dead Dove (185)
            Guardian Angel (112)
            Holy Grail (184)
            Holy Mantle (313)
            Mitre (173)
            Rosary (72)
            Sworn Protector (363)
            The Halo (101)
        Spun Added with "spun": true to the following:
            Experimental Treatment (240)
            Growth Hormones (70)
            Roid Rage (14)
            Speed Ball (143)
            Synthoil (345)
            The Virus (13)
        Yes, Mother? Added with "yesmother": true to the following:
            Mom's Bottle Of Pills (102)
            Mom's Bra (39)
            Mom's Pad (41)
            Mom's Wig (217)
            Mom's Eye (55)
            Mom's Purse (139)
            Mom's Contact (110)
            Mom's Knife (114)
            Mom's Heels (30)
            Mom's Eyeshadow (200)
            Mom's Perfume (228)
            Mom's Key (199)
            Mom's Lipstick (31)
            Mom's Underwear (29)
            Mom's Coin Purse (195)
            Mom's Pearls (355)
        Super Bum Added with "superbum": true to the following:
            Bum Friend (144)
            Dark Bum (278)
            Key Bum (388)

### CODE REVIEW

**Comment by viroulep** [comment]
Hey,

I have a few comments on the PR, not really on the content as everything seems to be handled, but more on the form.
First of all there is three commits, and two of them are cancelling each other :)
I guess Hyphen won't want to merge that "as is" but I may be wrong.
Do you have the technical knowledge on how to fix that ?
If not, do you want/need help or are you fine with someone else doing it for you ?

For the other things I'm going to comment directly in the commit, that will be easier.
Same here, if you don't know what I'm talking about in these comments, do you and @Hyphen-ated mind if I cherry-pick the last commit and clean+"pythonify" it ?

File: options_default.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   "size_multiplier": 1.0,
   "text_color": "#FFFFFF",
   "trackerserver_authkey": "",
+  "trackerserver_twitch_id": "cc0x4xs7f13phbtgco2xeflil2pu2uy",
```
Comment (viroulep) [inline options_default.json]: This was removed by c07c0c856b9183a14116bdb2ae757e212bfdda42, the twitch id is no longer needed by the tracker.

File: src/game_objects/state.py [hunk 2]
```diff
@@ -30,6 +30,16 @@ def reset(self, seed):
         self.bosses = []
         self.player_stats = {}
         self.guppy_set = set()
```
Comment (viroulep) [inline src/game_objects/state.py:32]: There is a lot of code duplication ahead, here we should probably use a defaultdict of sets to gather them in one "transformations" dict.

File: src/game_objects/state.py [hunk 3]
```diff
@@ -161,6 +171,27 @@ def __add_stats_for_item(self, item):
         # Can the .get thing be actually false ?!
         if item_info.guppy:
             self.guppy_set.add(item)
+        # If this can make us beelzebub, check if we're beelzebub
```
Comment (viroulep) [inline src/game_objects/state.py]: We can avoid this code duplication by creating a "transformation_list" in `item.py`, just like we used a stat_list, and iterate through it to check if we need to add the item to a specific transformation set.
(this is also true for the others 'if' in the code which are related to transformations)

File: src/view_controls/view.py [hunk 4]
```diff
@@ -132,20 +129,12 @@ def handle_events(self):
                 elif event.key == K_F4 and pygame.key.get_mods() & KMOD_ALT:
                     return Event.DONE
                 elif event.key == K_c and pygame.key.get_mods() & KMOD_CTRL:
-                    # Debug function to write the state to a json file
```
Comment (viroulep) [inline src/view_controls/view.py:135]: Ouch, this looks like a revert of @Zamiell's ea12728ba700d1cc11c5a59c1c7f7b6a66143d17, and would remove the copy to clipboard feature.

**Comment by Zamiell** [comment]
This PR looks like it is based on an old version of the tracker. Can you rebase the PR on the latest version so that it doesn't kill my changes?

**Comment by BrandonNowlin** [comment]
I'll pull the latest version (I thought I did but I might have made a mistake) and fix the issues mentioned above. I appreciate the feedback and I think I can take care of it.

**Comment by Hyphen-ated** [comment]
pulling won't do anything; this code is already based on the latest version as far as git is concerned. the issue is that it looks like you erased some existing code. (maybe you copy pasted an old version of the whole file at some point?)

**Pull request closed without merging**